### DISCUSSION TITLE
Thai doesn't use periods, but I get warnings

### DISCUSSION THREAD

**Discussion by LibreShawn** [dotnet/ResXResourceManager#622] [Q&A]
I have a Thai translation of my English strings and I am currently getting 37 warnings for strings that don't end with period. As far as I understand from searching the web, Thai doesn't use periods. Is there a way to turn off specific warnings for specific languages? Or maybe the code should be modified to not flag that warning for Thai.

I have a similar issue with U+2026, the single character ellipsis. If I use that character instead of 3 periods (...) in a Korean translation, it flags it as an error.

**Comment by tom-englert** [comment]
Punctuation is checked here https://github.com/dotnet/ResXResourceManager/blob/3122c9f3c2847714c45171a7acb2aa9f1d474744/src/ResXManager.Model/ResourceTableEntryRulePunctuation.cs#L13-L25

Maybe an additional configuration to exclude some languages from the checks at all would be fine for this?

Can you provide a PR for this?

**Reply by LibreShawn** [reply]
I have forked the repo and will take a look. I assume that adding a configuration for excluding languages would need a UI update for the configuration page? Also, what code would I modify to store that configuration info?

**Reply by LibreShawn** [reply]
I added an ExcludedCultures property to each rule and an AllExcludeCultures property to the ConfigurationEditor ViewModel. Then I modified all the CompliesToRule(s) methods to pass in the cultures along with the strings, and I modified the code to not check cultures in the ExcludedCultures list. That all seems to be working.

I modified the configuration UI to allow the input of a list of languages for each rule using the same mechanism used for the target languages in the translation view. Unfortunately, I'm not enough of a XAML expert to get that to work. The first time I go to the configuration tab, I can select languages and the UI seems to work. But those selections don't seem to carry over when I move back to the Main tab. Then when I go back to the Configuration tab, all my selections are gone and selecting languages doesn't work anymore.

I'm not sure how to proceed from here. Can you take a look at my UI code?

My code is here: https://github.com/LibreShawn/ResXResourceManager/tree/punctuation_exclude_cultures

**Comment by tom-englert** [comment]
The [ResourceTableEntryRules](https://github.com/dotnet/ResXResourceManager/blob/master/src/ResXManager.Model/ResourceTableEntryRules.cs) is serialized into the settings, actually it has only one data member, which is the `EnabledRuleIds` property.

However I wonder if adding all this overhead allowing such a detailed configuration for such an edge case is worth the effort, since it has to be tested and then maintained forever.

Maybe just turn of all checks for Thai, or a list of configurable languages?
